include/baresip.h

```diff
@@ -866,10 +866,25 @@ enum ua_event {
 	UA_EVENT_MODULE,
 	UA_EVENT_END_OF_FILE,
 	UA_EVENT_CUSTOM,
+	UA_EVENT_SIP_INCOMING,
 
 	UA_EVENT_MAX,
 };
 
+
+struct event {
+	enum ua_event ev;
+	const char *txt;
+	int err;
+	union {
+		struct ua *ua;
+		struct call *call;
+		const struct sip_msg *msg;
+		void *arg;
+	} u;
```

Comment:
union means that only one of the pointers are valid, for example ua OR call.
you cannot cast a ua pointer to a call pointer, or msg pointer etc.

is it better to use a struct here ?

most of the events have a valid UA pointer

Reply:
I also thought about the choice between union and struct. And decided to try first a union. Reasons for that:
- In the current UA event API we often pass a `call` and its `ua` which is redundant.
- Then we had often UA events where `ua` was set, but `call` was NULL.
- The new events which add a `sip_msg`  pointer should be used before a call is allocated.

It seems to work with a union. An application would have no benefit to have multiple of these pointers.